src/util/sandbox.cpp

```diff
@@ -369,80 +370,110 @@ QString Sandbox::migrateOldSettings() {
-            QLatin1String(
-                    "/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support/Mixxx");
+    // Parent path for Mixxx specific sandboxed macOS applications settings
+    QString sandboxAppSettingsPath = homePath +
+        QLatin1String("/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support");
+
+    // Legacy non-sandboxed mixxx settings path
+    QString legacySettingsPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
+    // Sandboxed mixxx settings path
+    QString sandboxedPath = sandboxAppSettingsPath + "/Mixxx";
     QDir sandboxedSettings(sandboxedPath);
 
+    // Ensure sandboxed parent directory for application settings exists even when not running
+    // in sandbox
+    QDir sandboxAppSettings(sandboxAppSettingsPath);
+    if (!sandboxAppSettings.exists()) {
+        if (!sandboxAppSettings.mkpath(sandboxAppSettingsPath)) {
+            qWarning() << "ERROR creating sandboxed application data directory"
+                       << sandboxAppSettingsPath;
+        }
+    }
+
     if (sandboxedSettings.exists() && !sandboxedSettings.isEmpty()) {
         return sandboxedPath;
     }
 
+    // Sandbox::askForAccess cannot be used here because it depends on settings being
+    // initialized. There is no need to store the bookmark anyway because this is a
+    // one time process.
+#ifdef __APPLE__
+
     // Because Mixxx cannot test if the old path exists before getting permission to access it
     // outside the sandbox, unfortunately it is necessary to annoy the user with this popup
     // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
     QString title = QObject::tr("Upgrading old Mixxx settings");
-    QString oldPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
     QMessageBox::information(nullptr,
             title,
             QObject::tr(
-                    "Due to Mac sandboxing, Mixxx needs your permission to "
-                    "access your music library "
-                    "and settings from Mixxx versions before 2.3.0. After "
-                    "clicking OK, you will see a file picker. "
-                    "To give Mixxx permission, press the Ok button in the file "
-                    "picker."
+                    "Existing Mixxx settings directory must be moved to new location."
                     "\n\n"
-                    "If you do not want to grant Mixxx access click Cancel on "
-                    "the file picker."));
+                    "Due to macOS sandboxing, Mixxx needs your permission to access your "
+                    "music library and move settings to the sandbox from Mixxx versions "
+                    "before 2.3.0. After clicking OK, you will see standard file open dialog. "
+                    "\n\n"
+                    "Select Open button in the file open dialog to grant Mixxx access to old files."
```

Comment:
```suggestion
                    "before 2.3.0. After clicking OK, you will see standard file selection dialog. "
                    "\n\n"
                    "Select Open button in the file selection dialog to grant Mixxx access to old files."
```
Using "file open dialog" and "Open button" so close together might be confusing.

Reply:
ping